provisioning/virtual-machines/ansible/roles/ubuntu-pre/tasks/main.yml

```diff
@@ -38,6 +38,23 @@
   apt:
     upgrade: 'yes'
 
+# tasks for installing desktop environment
+- name: Prevent installation of gdm3
+  dpkg_selections:
+    name: gdm3
+    selection: hold
+  when: desktop_mode is defined and desktop_mode == "true"
+
+- name: Install xubuntu-desktop
+  apt:
+    name: xubuntu-desktop
+    state: present
+  when: desktop_mode is defined and desktop_mode == "true"
```

Comment:
Should pay attention, because `Remove many unnecessary packages` is removing many packages this step reinstalls. You should probably mark them to prevent their reinstallation.